addons/account/models/account_journal_dashboard.py

```diff
@@ -374,7 +375,7 @@ def open_action(self):
         if not self.env.context.get('use_domain'):
             ctx['search_default_journal_id'] = self.id
         action['context'] = ctx
-        action['domain'] = self._context.get('use_domain', [])
+        action['domain'] = expression.AND([self._context.get('use_domain', []), [('journal_id', '=', self.id)]])
```

Comment:
@tbe-odoo I don't understand: the journal is already supposed to be in the contextual default search... why is it not working? what's context.get('use_domain') ? 

I'd rather do 
``` diff
        ctx.update({
            'journal_type': self.type,
            'default_journal_id': self.id,
            'default_type': invoice_type,
            'type': invoice_type,
+           'search_default_journal_id': self.id,
        })
```